### DISCUSSION TITLE
Weighing the performance of a single class/entity when training a NER model?

### DISCUSSION THREAD

**Discussion by Tadusko** [explosion/spaCy#8258] [Help: Coding & Implementations] [answered]
I'm training a custom NER model on a general training set, but I'm really only interested in tagging Locations. That's why I'd like to give the LOC tag more importance in the training process. **Is there a way to weigh the performance metrics of a single NER class?**

Looking at the config file, there's a section for ents_per_type, which by default is null.
```
[training.score_weights]
ents_per_type = null
ents_f = 1.0
ents_p = 0.0
ents_r = 0.0
```
After training, ents_per_type houses a nested dictionary in the meta.json with performance scores for each entity. For example:
```
    "ents_per_type":{
      "ORG":{
        "p":0.8813559322,
        "r":0.870292887,
        "f":0.8757894737
      }
...
```
My intuition is that this could be used, but didn't find any examples online. The dictionary structure makes this one different than e.g. ents_f, which is simply a float. 

I tried pointing to the F-score of the LOC tag. While I could train without errors, I didn't see a performance increase, so I assume this doesn't work:
`[training.score_weights]`
`ents_per_type["LOC"]["f"] = 0.6`

Spacy ver.: 3.0.5

**Comment by svlandeg** [comment, marked as answer]
```
[training.score_weights]
ents_per_type["LOC"]["f"] = 0.6
```
This doesn't currently work - though I like the suggestion. I'm afraid there's currently no straightforward way to weigh by class.

If you're only interested in `Location` entities, you could pre-filter your training data though? Or artificially boost (up-sample) the number of `Location` examples in your data?

**Reply by Tadusko** [reply]
Ah, too bad. Thank you for the answer and the suggestions!
